File: Suiteva.API/Models/SeasonalRate.cs
using System.ComponentModel.DataAnnotations;

namespace Suiteva.API.Models;

public class SeasonalRate
{
    public int Id { get; set; }
    
    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;
    
    public DateTime StartDate { get; set; }
    
    public DateTime EndDate { get; set; }
    
    public decimal Multiplier { get; set; }
    
    public int HotelId { get; set; }
    public Hotel? Hotel { get; set; }
}

File: suiteva-client/src/app/services/seasonal-rate.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models';

export interface SeasonalRateDto {
    id: number;
    name: string;
    startDate: string;
    endDate: string;
    multiplier: number;
    hotelId: number;
}

export interface CreateSeasonalRateDto {
    name: string;
    startDate: string;
    endDate: string;
    multiplier: number;
    hotelId: number;
}

@Injectable({
    providedIn: 'root'
})
export class SeasonalRateService {
    private readonly apiUrl = '/api/SeasonalRates';

    constructor(private readonly http: HttpClient) { }

    getRatesByHotel(hotelId: number): Observable<ApiResponse<SeasonalRateDto[]>> {
        return this.http.get<ApiResponse<SeasonalRateDto[]>>(`${this.apiUrl}/${hotelId}`);
    }

    createRate(data: CreateSeasonalRateDto): Observable<ApiResponse<SeasonalRateDto>> {
        return this.http.post<ApiResponse<SeasonalRateDto>>(this.apiUrl, data);
    }

    deleteRate(id: number): Observable<ApiResponse<boolean>> {
        return this.http.delete<ApiResponse<boolean>>(`${this.apiUrl}/${id}`);
    }
}
